workflow/validate/validate.go

```diff
@@ -1201,6 +1203,9 @@ func (ctx *templateValidationCtx) validateDAG(scope map[string]interface{}, tmpl
 		// add all tasks outputs to scope so that a nested DAGs can have outputs
 		prefix := fmt.Sprintf("tasks.%s", task.Name)
 		ctx.addOutputsToScope(resolvedTmpl, prefix, scope, false, false)
+		if task.Hooks != nil && task.Hooks.Exit != nil {
```

Comment:
do we need to guard this? if so, use `HasExitHook`

Reply:
yes, Including the current step/task variables should be only for `exithook`.